### PR TITLE
Fixes promote action on main branch

### PR DESCRIPTION
**Pull request by saltiyazan**
# Description

Improves the promote workflow.
Allows the workflow to promote charm only from the current branch, in this case main.

## Checklist

- [x] My code follows the [style guidelines](/CONTRIBUTING.md) of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have made corresponding changes to the documentation.
- [ ] I have added tests that validate the behaviour of the software.
- [x] I validated that new and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I have bumped the version of any required library.

### CODE REVIEW

File: .github/workflows/charms.promote.yaml [hunk 1]
```diff
@@ -105,7 +105,7 @@ jobs:
             echo "promote-to=stable" >> ${GITHUB_ENV}
           fi
       - name: Promote all Charms
-        uses: canonical/charming-actions/release-charm@1.0.3
+        uses: canonical/charming-actions/release-charm@2.2.2
```
Comment (Gmerold) [inline .github/workflows/charms.promote.yaml:108]: Don't we wanna go with the `latest` in the `main` branch?

Comment (sanchezfdezjavier) [reply .github/workflows/charms.promote.yaml:108]: What do we mean with `latest`? If we're taking about the charming actions, that's the latest version, doesn't it?

Comment (Gmerold) [reply .github/workflows/charms.promote.yaml:108]: I mean `release-charm@latest`

Comment (saltiyazan) [reply .github/workflows/charms.promote.yaml:108]: valid point. Done

**Review by Gmerold — APPROVED** [review]

**Review by ghislainbourgeois — APPROVED** [review]

**Review by sanchezfdezjavier — APPROVED** [review]

File: .github/workflows/charms.promote.yaml (hunk 1 above)
Comment (saltiyazan) [reply .github/workflows/charms.promote.yaml:108]: @latest doesn't work. Added back the version numbers.

**Pull request merged**